service/debugger/debugger.go

```diff
@@ -1205,6 +1208,16 @@ func (d *Debugger) Command(command *api.DebuggerCommand, resumeNotify chan struc
 
 	if err != nil {
 		var errProcessExited proc.ErrProcessExited
+
+		// Process exited so we should "re-foreground" the main process, so that it can capture interrupts
+		if errors.As(err, &errProcessExited) && !tcxpgrp.IsForeground() {
+			pgid := syscall.Getpgrp()
```

Comment:
The body of this branch can be folded into the branch below, I think.

Also, this functionality must be encapsulated somewhere else in a unix specific file, with another implementation available for Windows (can be stubbed initially if you don't plan to fix this for Windows in this PR). Can you place place this code in a function maybe within `pkg/terminal/` somewhere with proper build tags via file name for unix/windows, might even be worth creating another small subpackage of terminal, but perhaps not. Then you can import and call here.

The reason is this layer of the code is *supposed* to be platform agnostic as much as possible.